File: java/src/main/java/com/languageweaver/sdk/samples/languagedetection/text/TextLanguageDetectionService.java
package com.languageweaver.sdk.samples.languagedetection.text;

import com.languageweaver.sdk.common.LanguageWeaverClient;
import com.languageweaver.sdk.common.SdkFactory;
import com.languageweaver.sdk.common.configurations.ClientConfiguration;
import com.languageweaver.sdk.common.constants.Format;
import com.languageweaver.sdk.languagedetection.common.request.TextLanguageDetectionRequest;
import com.languageweaver.sdk.languagedetection.common.result.LanguageDetectionResult;

public class TextLanguageDetectionService {
    public static void main(String[] args) throws Exception {
        try (LanguageWeaverClient lwClient = new SdkFactory().getLanguageWeaverClient(new ClientConfiguration())) {
            TextLanguageDetectionRequest textLanguageDetectionRequest = new TextLanguageDetectionRequest()
                    .setInput("A mountain is generally steeper than a hill")
                    .setInputFormat(Format.PLAIN);

            final LanguageDetectionResult languageDetectionResult = lwClient.languageDetectionText(textLanguageDetectionRequest);

            if (languageDetectionResult.getLanguages() != null) {
                languageDetectionResult.getLanguages()
                        .forEach(language -> System.out.println(language.getName()));
            }
        }
    }
}

File: javascript/browser/src/samples/language-detection/text/TextLanguageDetectionService.js
import SdkFactory, {
    ClientConfiguration,
    CredentialsConfiguration,
    Format,
    TextLanguageDetectionRequest,
} from "@language-weaver/lw-sdk-js";

export const detectTextLanguage = async (clientId, clientSecret) => {
    try {
        const clientConfiguration = new ClientConfiguration();
        clientConfiguration.credentialsConfiguration = new CredentialsConfiguration(clientId, clientSecret);
        const lwClient = await SdkFactory.getLanguageWeaverClient(clientConfiguration);
        const textLanguageDetectionRequest = new TextLanguageDetectionRequest();
        textLanguageDetectionRequest.input = "A mountain is generally steeper than a hill";
        textLanguageDetectionRequest.inputFormat = Format.PLAIN;

        const textLanguageDetectionResult = await lwClient.detectTextLanguage(textLanguageDetectionRequest);
        console.log(textLanguageDetectionResult);
    } catch (e) {
        console.log(e);
    }
}